File: backend/tiggle-root/tiggle/src/main/java/com/side/tiggle/domain/comment/dto/CommentDto.java
package com.side.tiggle.domain.comment.dto;

import com.side.tiggle.domain.comment.model.Comment;
import com.side.tiggle.domain.member.model.Member;
import com.side.tiggle.domain.transaction.model.Transaction;
import lombok.Getter;
import lombok.Setter;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;


@Getter
@Setter
public class CommentDto {

    @NotNull
    Long txId;
    Long parentId;
    @NotNull
    Long senderId;
    @NotNull
    Long receiverId;
    @NotNull @Size(max = 255)
    String content;

    public Comment toEntity(Transaction tx, Member sender, Member receiver) {
        return Comment.builder()
                .tx(tx)
                .parentId(parentId)
                .receiver(receiver)
                .sender(sender)
                .content(content)
                .build();
    }

    public static CommentDto fromEntity(Comment comment) {
        CommentDto dto = new CommentDto();
        dto.txId = comment.getTx().getId();
        dto.parentId = comment.getParentId();
        dto.senderId = comment.getSender().getId();
        dto.receiverId = comment.getReceiver().getId();
        dto.content = comment.getContent();

        return dto;
    }
}

File: backend/tiggle-root/tiggle/src/main/kotlin/com/side/tiggle/domain/comment/dto/CommentDto.kt
package com.side.tiggle.domain.comment.dto

import com.side.tiggle.domain.comment.model.Comment
import com.side.tiggle.domain.member.model.Member
import com.side.tiggle.domain.transaction.model.Transaction
import javax.validation.constraints.Size

open class CommentDto(
    val txId: Long,
    val parentId: Long?,
    var senderId: Long,
    val receiverId: Long,
    @Size(max = 255)
    val content: String
) {

    fun toEntity(tx: Transaction, sender: Member, receiver: Member): Comment{
        return Comment(
            tx = tx,
            parentId = parentId,
            receiver = receiver,
            sender = sender,
            content = content
        )
    }
    companion object {
        fun fromEntity(comment: Comment): CommentDto {
            return CommentDto(
                txId = comment.tx.id,
                parentId = comment.parentId,
                senderId = comment.sender.id,
                receiverId = comment.receiver.id,
                content = comment.content
            )
        }
    }
}